packages/cli/generators/datasource/index.js

```diff
@@ -80,6 +80,12 @@ module.exports = class DataSourceGenerator extends ArtifactGenerator {
   promptArtifactName() {
     debug('Prompting for artifact name');
     if (this.shouldExit()) return false;
+
+    if (this.options.name) {
+      Object.assign(this.artifactInfo, {name: this.options.name});
+      return;
+    }
```

Comment:
@achrinza  I have updated the code but we still need that check for artifactInfo.name because datasource generator does not use the base prompt and has its own implementation.